types/time.go

```diff
@@ -823,12 +823,16 @@ func parseDatetime(sc *stmtctx.StatementContext, str string, fsp int8, isFloat b
 
 	tmp := FromDate(year, month, day, hour, minute, second, microsecond)
 	if overflow {
-		// Convert to Go time and add 1 second, to handle input like 2017-01-05 08:40:59.575601
-		t1, err := tmp.GoTime(gotime.Local)
-		if err != nil {
+		if sc != nil {
```

Comment:
When will `sc` be nil?

Reply:
When you invoke function, pass the nil to function.
You can find some examples in my pr code.

Reply:
Can we make sure the code does not pass a `nil` session context and remove this check?

Reply:
You mean we no need check sc in here. And just modify our code don't pass nil to it?

Reply:
Yes, is that possible?

Reply:
It should be possible, after all it sounds meaningless for a `nil` context. 

I found there are a lot of test codes that passed `nil`, it would be great if you can fix it. @Rustin-Liu

Reply:
I will  fix it tonight.